keps/sig-network/4974-deprecate-endpoints/README.md

```diff
@@ -0,0 +1,532 @@
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+- <test>: <link to test coverage>
+
+##### e2e tests
+
+We will want to add a new periodic e2e job that confirms that the e2e
+suite passes in a cluster with the Endpoints controllers disabled.
+This will require also adding a Feature tag to allow skipping the
+Endpoints-specific tests.
+
+There are quite a few places in the e2e tests that currently use
+`v1.Endpoints`:
+
+  - Many of the tests in `test/e2e/network/service.go` do various
+    checks on both Endpoints and EndpointSlices. This will need to be
+    split into separate tests, with the Endpoints tests
+    feature-tagged.
+
+  - Some of the tests in `test/e2e/network/endpointslice.go` will need
+    to be split up, to separately test "EndpointSlices are created
+    correctly" and "EndpointSlices match Endpoints" in separate tests,
+    with the Endpoints tests feature-tagged.
+
+  - The conformance tests in
+    `test/e2e/network/endpointslicemirroring.go` should be demoted
+    from conformance, and all of the tests should be feature-tagged,
+    but should otherwise be unchanged.
+
+  - Several tests in `test/e2e/network/dual_stack.go` check that the
+    Endpoints controller does the right thing but _do not_ check that
+    the EndpointSlice controller does the right thing (which means
+    that we do not actually have any proper e2e testing of dual-stack
+    Services). These should be updated to test EndpointSlices, with
+    the Endpoints tests split out into separate feature-tagged tests.
+
+  - `[It] [sig-network] Services should test the lifecycle of an
+    Endpoint [Conformance]`: This just tests that the API works and
+    does not test the behavior of the controllers, so it just needs to
+    be feature-tagged, but not changed.
+
+  - `[It] [sig-network] Service endpoints latency should not be very
+    high [Conformance]`: This tests the latency of the Endpoints
+    controller, and should be fixed to test the latency of the
+    EndpointSlice controller instead, since the latency of the
+    Endpoints controller has no impact on the functioning of a
+    cluster.
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+- <test>: <link to test coverage>
+
+### Graduation Criteria
+
+#### Alpha
+
+- All Endpoints and EndpointSlice Mirroring tests have been demoted
+  from conformance, and we document this (e.g. via a blog post).
```

Comment:
Ideally, try to cover this in the "mid point comms" blog about the upcoming release that will deprecate Endpoints.